smartpy_arc/arc_pandas.py

```diff
@@ -137,11 +137,11 @@ def arc_to_pandas(workspace_path, class_name, index_fld=None, flds=None, spatial
             df.sort_index(inplace=True)
 
         # set nulls back if desired
-        # TODO: look possible make this the defaualt and
+        # TODO: look possible make this the default and
         # use more distinct null values
         if not fill_nulls:
-            df.replace([num_fill, str_fill, date_fill, 'nan'], np.nan, inplace=True)
-
+            df.replace([num_fill, str_fill, 'nan'], np.nan, inplace=True)
+            df = df.astype(str).replace(date_fill, np.nan)
```

Comment:
I'm not sure about this line, seems like it would convert the whole dataframe to strings? Maybe I need to re-think all these null issues.

Reply:
Oh yeah.. duh.... it is definitely converting all columns to string.

Reply:
Something like this is pretty ugly, but maybe?

```
for column in df.select_dtypes(include=['datetime64[ns]']):
    df[column] = pd.to_datetime(df[column].astype(str).replace('1678-01-01 00:00:00', np.nan))
```